test/lib/mergeArrayByName.test.js

```diff
@@ -0,0 +1,76 @@
+const branchArrayMerge = require('../../lib/mergeArrayByName')
+const YAML = require('js-yaml')
+
+describe('mergeArrayByName', () => {
+  it('works', () => {
+    const target = [{
+      name: 'master',
+      shouldChange: 'did not change',
+      shouldKeep: 'kept'
+    }, {
+      name: 'develop'
+    }]
+
+    const source = [{
+      name: 'master',
+      shouldChange: 'did change'
+    }, {
+      name: 'added'
+    }]
+
+    const merged = branchArrayMerge(target, source)
+
+    expect(merged).toEqual([{
+      name: 'master',
+      shouldChange: 'did change',
+      shouldKeep: 'kept'
+    }, {
+      name: 'develop'
+    }, {
+      name: 'added'
+    }])
+  })
+
+  it('works in a realistic scenario', () => {
+    const target = YAML.load(`
```

Comment:
One last thing and then we should be good - internally, everything uses `YAML.safeLoad`, so we should probably use that in our tests too for maximum consistency.

```suggestion
    const target = YAML.safeLoad(`
```